File: PSP/graphql/backend/src/main/java/com/example/demo/data/repositories/VisitasRepository.java
package com.example.demo.data.repositories;

import com.example.demo.data.modelo.VisitaEntity;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.ListCrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface VisitasRepository extends ListCrudRepository<VisitaEntity, Long>{


    List<VisitaEntity> findAllByUser_Name(String userName);


    @Query("""
             SELECT v FROM VisitaEntity v
             JOIN FETCH v.pois p
             JOIN FETCH p.ciudad c
             WHERE v.user.name = :userName
             """)
    List<VisitaEntity> findAllByUser_NameWithAll(String userName);



}

File: PSP/graphql/frontend/src/main/kotlin/org/example/frontend/data/VisitasRepository.kt
package org.example.frontend.data

import com.apollographql.apollo3.ApolloClient
import org.example.visitas.VisitasQuery
import org.springframework.stereotype.Repository

@Repository
class VisitasRepository(
    var apolloClient: ApolloClient
) {

    suspend fun getVisitas() = apolloClient.query(VisitasQuery()).execute().data?.getVisitas

}
